apps/sms/test/marionette/lib/messages.js

```diff
@@ -237,13 +237,21 @@
               break;
             }
           }
-
-          client.helper.waitForElementToDisappear(menuElement);
         },
 
         addRecipient: function(number) {
           this.Composer.recipientsInput.sendKeys(number + Chars.ENTER);
 
+          // Since recipient.js re-renders recipients all the time (when new
+          // recipient is added or old is removed) and it can happen several
+          // times (!!!) during single "add" or "remove" operation we should
+          // wait until Recipients View is in a final state. The problem here is
+          // that between "findElement" and "displayed" calls element can
+          // actually be removed from DOM and re-created again that will lead to
+          // "stale element" exception. Eventually real fix should be introduced
+          // in recipients.js to avoid wasteful rendering.
+          client.helper.wait(500);
```

Comment:
Intermittent failures might probably happen due to the 500 ms waiting. Something the test might need more than 500 ms to wait for running correctly.

Looks like we could use `searchTimeout`[1] to avoid this issue.

[1]: https://github.com/mozilla-b2g/gaia/blob/master/apps/calendar/test/marionette/lib/calendar.js#L16